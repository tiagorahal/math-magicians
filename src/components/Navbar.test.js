import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Navbar from './navbar';

it('matches the snapshot', () => {
  const tree = renderer
    .create(<Navbar />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders the main heading', () => {
  render(<Navbar />);
  const headingElement = screen.getByText(/^Math Magicians$/i);

  expect(headingElement).toBeInTheDocument();
});

it('renders the navigation links', () => {
  render(<Navbar />);
  const homeLinkElement = screen.getByText(/Home/i);
  const calculatorLinkElement = screen.getByText(/Calculator/i);
  const quoteLinkElement = screen.getByText(/Quote/i);

  expect(homeLinkElement).toBeInTheDocument();
  expect(calculatorLinkElement).toBeInTheDocument();
  expect(quoteLinkElement).toBeInTheDocument();
});
