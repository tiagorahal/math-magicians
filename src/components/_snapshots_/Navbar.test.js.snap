// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <div
    className="navbar-component"
  >
    <h1>
      Math Magicians
    </h1>
    <nav
      className="navbar"
    >
      <ul
        className="navbar-list"
      >
        <li
          className="navbar-list-items"
        >
          <a
            aria-current="page"
            className="active-link"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <span
            className="pipe-divisor"
          >
             | 
          </span>
        </li>
        <li
          className="navbar-list-items"
        >
          <a
            aria-current={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
          <span
            className="pipe-divisor"
          >
             | 
          </span>
        </li>
        <li
          className="navbar-list-items"
        >
          <a
            aria-current={null}
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
          <span
            className="pipe-divisor"
          >
             | 
          </span>
        </li>
      </ul>
    </nav>
  </div>,
  <h2>
    Welcome Math Magicians!
  </h2>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
  </p>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
  </p>,
]
`;